### PR TITLE
[catalog] Selecting one atlas server if there are multiple listed out…

### PR DESCRIPTION
**Pull request by quadoss**
Selecting one atlas server if there are multiple listed out under atlas.rest.address

without the changes will see the following exceptions

```
[23/Mar/2021 16:20:14 -0700] atlas_client INFO     Atlas server to use: ['https://nightly7x-2.nightly7x.root.hwx.site:31443/', 'https://nightly7x-1.nightly7x.root.hwx.site:31443/']
[23/Mar/2021 16:20:14 -0700] catalog_api  ERROR    No connection adapters were found for u"['https://nightly7x-2.nightly7x.root.hwx.site:31443/', 'https://nightly7x-1.nightly7x.root.hwx.site:31443/']'/api/atlas/v2/types/typedefs?type=classification'
Traceback (most recent call last):
  File "/opt/cloudera/parcels/CDH-7.2.7-1.cdh7.2.7.p0.9408337/lib/hue/desktop/libs/metadata/src/metadata/catalog_api.py", line 55, in decorator
    return view_fn(*args, **kwargs)
  File "/opt/cloudera/parcels/CDH-7.2.7-1.cdh7.2.7.p0.9408337/lib/hue/desktop/libs/metadata/src/metadata/catalog_api.py", line 230, in list_tags
    data = api.search_entities_interactive(facetFields=['tags'], facetPrefix=prefix, limit=limit, offset=offset)
  File "/opt/cloudera/parcels/CDH-7.2.7-1.cdh7.2.7.p0.9408337/lib/hue/desktop/libs/metadata/src/metadata/catalog/atlas_client.py", line 226, in search_entities_interactive
    classification_response = self._root.get('/v2/types/typedefs?type=classification')
  File "/opt/cloudera/parcels/CDH-7.2.7-1.cdh7.2.7.p0.9408337/lib/hue/desktop/core/src/desktop/lib/rest/resource.py", line 157, in get
    return self.invoke("GET", relpath, params, headers=headers, allow_redirects=True, clear_cookies=clear_cookies)
  File "/opt/cloudera/parcels/CDH-7.2.7-1.cdh7.2.7.p0.9408337/lib/hue/desktop/core/src/desktop/lib/rest/resource.py", line 87, in invoke
    log_response=log_response
  File "/opt/cloudera/parcels/CDH-7.2.7-1.cdh7.2.7.p0.9408337/lib/hue/desktop/core/src/desktop/lib/rest/resource.py", line 114, in _invoke
    clear_cookies=clear_cookies
  File "/opt/cloudera/parcels/CDH-7.2.7-1.cdh7.2.7.p0.9408337/lib/hue/desktop/core/src/desktop/lib/rest/http_client.py", line 229, in execute
    raise self._exc_class(ex)
RestException: No connection adapters were found for u"['https://nightly7x-2.nightly7x.root.hwx.site:31443/', 'https://nightly7x-1.nightly7x.root.hwx.site:31443/']/api/atlas/v2/types/typedefs?type=classification"
```

Post the fix we would see these lines with LOG.info added into the atlas_client:
```

[23/Mar/2021 16:24:22 -0700] atlas_client INFO     Atlas server to use: ['https://nightly7x-2.nightly7x.root.hwx.site:31443/', 'https://nightly7x-1.nightly7x.root.hwx.site:31443/']
[23/Mar/2021 16:24:22 -0700] atlas_client INFO     Atlas server to use: https://nightly7x-2.nightly7x.root.hwx.site:31443/
```

### CODE REVIEW

File: desktop/libs/metadata/src/metadata/catalog/atlas_client.py [hunk 1]
```diff
@@ -65,7 +65,9 @@ class AtlasApi(Api):
   def __init__(self, user=None):
     super(AtlasApi, self).__init__(user)
 
-    self._api_url = CATALOG.API_URL.get().strip('/') + "/api/atlas"
+    # selecting the first atlas server from the list of server if there are multiple server with comma seperated
+    atlas_server_to_use = CATALOG.API_URL.get().split(',')[0].replace("'","").replace("[","")
```
Comment (romainr) [inline desktop/libs/metadata/src/metadata/catalog/atlas_client.py:69]: Could we add some quick tests?

**Comment by quadoss** [comment]
This is how we receive the url if in HA mode.  requests works regardless of https/http as it just checks to see if atlas is Active or Passive - no login needed for status check.
['https://<hostname1>:31443/',%20'https://<hostname2>:31443/']

File: desktop/libs/metadata/src/metadata/catalog/atlas_client.py [hunk 2]
```diff
@@ -65,7 +66,25 @@ class AtlasApi(Api):
   def __init__(self, user=None):
     super(AtlasApi, self).__init__(user)
 
-    self._api_url = CATALOG.API_URL.get().strip('/') + "/api/atlas"
+    # Checking which server is active server if there are multiple else use the one thats listed
+    atlas_servers = CATALOG.API_URL.get().replace("%20 ","").replace("['","").replace("']","").replace("'","").split(',')
```
Comment (nightscape) [inline desktop/libs/metadata/src/metadata/catalog/atlas_client.py:70]: Circle CI complains about missing whitespace after the comma, and I think the whitespace in `"%20 "` should not be there (at least it isn't there in our case, we're facing the same issue).
The full line would be
```python
atlas_servers = CATALOG.API_URL.get().replace("%20", "").replace("['", "").replace("']", "").replace("'", "").split(',')
```

**Comment by nightscape** [comment]
@romainr except for the failing CircleCI (which is about whitespace and can easily be fixed), is there anything else left in order to get this merged, and hopefully released in a CDH patch upgrade?

**Review by Akhilsnaik — APPROVED** [review]
LGTM

**Pull request merged**